### ISSUE TITLE
imports-loader doesn't generate source maps by default

### ISSUE THREAD

**Issue by benjamin-hodgson** [webpack-contrib/imports-loader#111]
### Actual Behavior

My `rules` config looks like this:

```javascript
{
    rules: [
        {
            test: /\.jsx?$/,
            use: [{
                loader: "imports-loader",
                options: {
                    additionalCode: 'console.log("hello");'
                }
            }],
            exclude: /node_modules/,
        }
    ]
}
```

When I bundle a JS file using this config and then look at the source-mapped output in the Chrome debugger, the code injected by `imports-loader` is visible at the top of the file:

```javascript
/*** IMPORTS FROM imports-loader ***/


console.log("hello");
```

### Expected Behavior

The code injected by `imports-loader` should not be visible in the source mapped output.

### How Do We Reproduce?

1. Set up a webpack config with source maps enabled (eg `devTool: "inline-source-map"`).
2. Set up an `imports-loader` with `test: /\.jsx?$/`
3. Bundle a single `.js` file
4. Look at the source-mapped version of that file in the Chrome dev tools

### Other notes

* Source mapping works correctly when using `imports-loader` alongside other loaders which generate source maps (eg `ts-loader`)
* The [existing test](https://github.com/webpack-contrib/imports-loader/blob/master/test/loader.test.js#L326) doesn't detect the bug because it's plugged in `babel-loader` before the `imports-loader`
* I think the bug is in [this `if` statement](https://github.com/webpack-contrib/imports-loader/blob/master/src/index.js#L77) - it needs an `else if (this.sourceMap)` block in which we create a new sourcemap.

**Comment by alexander-akait** [comment]
```
{"version":3,"file":"main.js","mappings":";;;;;AAAA;;AAEA","sources":["webpack://imports-loader-bug/./src/index.js"],"sourcesContent":["/*** IMPORTS FROM imports-loader ***/\n\nconsole.log('hello');\n\n"],"names":[],"sourceRoot":""}
```

All works fine

**Comment by benjamin-hodgson** [comment]
Did you actually try to use the sourcemap in the dev tools?

**Comment by benjamin-hodgson** [comment]
I promise you this is a real bug. The existing tests don't catch it because they are not reproducing the bug, not because the bug isn't there.

Here is a failing test:

```javascript
  it('should work with source maps when the "devtool" option is enabled', async () => {
    const compiler = getCompiler(
      "some-library.js",
      {},
      {
        devtool: "source-map",
        module: {
          rules: [
            {
              test: /\.js$/i,
              rules: [
                {
                  loader: path.resolve(__dirname, "../src"),
                  options: { imports: "lib_1" },
                },
              ],
            },
          ],
        },
      }
    );
    const stats = await compile(compiler);
    const bundle = readAsset("main.bundle.js", stats).split("\n");
    const sourceMap = readAsset("main.bundle.js.map", stats);

    const consumer = new SourceMapConsumer(sourceMap);
    const result = consumer.generatedPositionFor({
      line: 1,
      column: 0,
      source: "webpack://ImportsLoader/some-library.js",
    });
    expect(bundle[result.line - 1 /* 1-indexed */]).not.toContain(
      "/*** IMPORTS FROM imports-loader ***/"
    );
  });
```

**Comment by alexander-akait** [comment]
I can't reproduce your problem using your example

**Comment by benjamin-hodgson** [comment]
Did you run the unit test from #112?

**Comment by benjamin-hodgson** [comment]
Fixed by #113
